java/client/src/main/java/glide/api/logging/Logger.java

```diff
@@ -0,0 +1,85 @@
+/** Copyright GLIDE-for-Redis Project Contributors - SPDX Identifier: Apache-2.0 */
+package glide.api.logging;
+
+import lombok.Getter;
+import lombok.NonNull;
+
+public class Logger {
+    // TODO: consider lazy loading the glide_rs library
+    static {
+        System.loadLibrary("glide_rs");
+    }
+
+    // Enum ordinal is used, so order of variants must be kept the same
+    @Getter
+    public enum Level {
+        DISABLED(-1),
+        ERROR(0),
+        WARN(1),
+        INFO(2),
+        DEBUG(3),
+        TRACE(4);
+
+        private final int level;
+
+        private Level(int level) {
+            this.level = level;
+        }
+    }
+
+    private static Logger instance;
+    private static Level loggerLevel;
+
+    private Logger(@NonNull Level level, String fileName) {
```

Comment:
nonnull for file

Reply:
Oh right, I forgot about that.
Consider using `Optional<String>` then.

Reply:
I hate Java

Reply:
Okay, I tried it and I get a bunch of warnings about how I should not be using Optionals as argument types. Seems like they're meant to be used for return types only, so I will leave it as is.